sonic-ledd/scripts/ledd

```diff
@@ -34,200 +35,88 @@ Options:
 
 LED_MODULE_NAME = "led_control"
 LED_CLASS_NAME = "LedControl"
+SELECT_TIMEOUT = 1000
 
-SONIC_CFGGEN = "/usr/local/bin/sonic-cfggen"
-MINIGRAPH_FILE = "/etc/sonic/minigraph.xml"
-HWSKU_KEY = "minigraph_hwsku"
-PLATFORM_KEY = "platform"
+class DaemonLedd(DaemonBase):
+    def __init__(self):
+        DaemonBase.__init__(self)
+
+    def __exit__(self):
+        DaemonBase.__exit__(self)
+
+    def run(self):
+        if not os.geteuid() == 0:
```

Comment:
I feel like this check for root user belongs at the top of the `main()` function, and not in the daemon class. @kevinwangsk: do you agree here?

Reply:
Agree.